### PR TITLE
feat: reworks Chrome/Edge driver download to avoid XML processing

### PR DESCRIPTION
**Pull request by jomarko**
#### Short description of what this resolves:
Changes in this PR tries to download browser webdrivers more straightforwardly. We do not parse xml with all versions available anymore, as we can construct the url for downloading particular webdriver without it.

If the webdriver is available, response has `200` status code.
If the webdriver is not available, response has `404` status code.

#### Changes proposed in this pull request:
`ChromeDriverBinaryHandler` and`EdgeDriverBinaryHandler` do not more extend `XmlStorageSource` to avoid xml file parsing. The binary url can be derived without xml parsing for these browsers.

**Fixes**:
- #309

### CODE REVIEW

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/UrlStorageSource.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    public ExternalBinary getReleaseForVersion(String requiredVersion) throws Exception {
+        return getReleaseForVersion(requiredVersion, Architecture.AUTO_DETECT);
+    }
+
+    @Override
+    public ExternalBinary getReleaseForVersion(String requiredVersion, Architecture architecture) throws Exception {
+        final String externalBinaryUrl = storageUrl + getFileNameRegexToDownload(requiredVersion, architecture);
+
+        final HttpClient.Response response = httpClient.get(externalBinaryUrl);
+        if (response.getStatusCode() == 404) {
```
Comment (bartoszmajsak) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/UrlStorageSource.java:83]: What if the server responds with e.g. `503` or even `403`? I would rather verify if we succeeded (so we got `200`) and return the binary but fail with all the other cases.

**Review by bartoszmajsak — COMMENTED** [review]
Excellent! Thank you very much. I have one question before we merge it.

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/UrlStorageSource.java (hunk 1 above)
Comment (jomarko) [reply drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/UrlStorageSource.java:83]: thank you, I will update the PR

**Review by bartoszmajsak — APPROVED** [review]
Great work @jomarko!

**Pull request merged**